src/transformers/generation/logits_process.py

```diff
@@ -474,12 +476,43 @@ def _calc_banned_ngram_tokens(
 
 class NoRepeatNGramLogitsProcessor(LogitsProcessor):
     r"""
-    [`LogitsProcessor`] that enforces no repetition of n-grams. See
+    N-grams are groups of "n" consecutive words, characters, or tokens taken from a sequence of text. Given a sentence:
+    " She runs fast ", the bi-grams (n = 2) would be ("she","runs") and ("runs","fast"). In text generation, avoiding
+    repetitions of word sequences provides a more diverse and coherent output. This [`LogitsProcessor`] enforces no
+    repetition of n-grams. See
     [Fairseq](https://github.com/pytorch/fairseq/blob/a07cb6f40480928c9e0548b737aadd36ee66ac76/fairseq/sequence_generator.py#L345).
+    It calls the <_calc_banned_ngram_tokens> function to determine the banned tokens (n-grams) based on the current
+    state of the generated sequences and the specified ngram_size. For each hypothesis, it sets the scores of banned
+    tokens to negative infinity (-inf), effectively discouraging the model from generating those tokens.
```

Comment:
```suggestion
```
These are implementation details, the user reading the docs doesn't need to know this. Shorter docs = user can find they need faster :)

Reply:
Got it. 👍